x/sync/client.go

```diff
@@ -108,7 +108,16 @@ func (c *client) GetChangeProof(ctx context.Context, req *pb.SyncGetChangeProofR
 			return nil, err
 		}
 
-		if err := db.VerifyChangeProof(ctx, &changeProof, req.StartKey, req.EndKey, endRoot); err != nil {
+		startKey := merkledb.Nothing[[]byte]()
+		if !req.StartKey.IsNothing {
+			startKey = merkledb.Some(req.StartKey.Value)
+		}
+		endKey := merkledb.Nothing[[]byte]()
+		if !req.EndKey.IsNothing {
+			endKey = merkledb.Some(req.EndKey.Value)
```

Comment:
Right now we assume that `req.StartKey` and `req.EndKey` are non-nil but we don't have to do that. Should we?

Reply:
I like the idea of putting these checks into the function. Seems less likely we miss adding this prelude somewhere...